### DISCUSSION TITLE
File type icons issue

### DISCUSSION THREAD

**Discussion by Ruygazunc** [sumatrapdfreader/sumatrapdf#4602] [General]
I would like Sumatra to be my default app for both Epub and Mobi, but I'd like to use different icons for each one. When I change the icon for Epub with FileTypesMan, the Mobi icons change as well, and vice versa. The only way I can get different icons is to disassociate Mobi from Sumatra. Anyone know a workaround here without installing another app just for Mobi files?

**Comment by GitHubRulesOK** [comment]
Hmm I think it was possible if you self compile the "debug" version as that allowed such changes in the past but  the whole area about "associations" is very much windows version related so what worked in the past may not work now 

When one app.EXE is associated with different "types.ext" the common EXE icon is used the exception here is that Edge uses a red icon for PDF rather than its native blue one.?


![image](https://github.com/user-attachments/assets/36bb64c7-8a53-4627-9f1e-500e75a5407b)


When installed sumatraPDF uses specifics for CBX and ePub but Mobi is the default.exe
![image](https://github.com/user-attachments/assets/e393cb5c-bd7f-446c-8efd-25c7262ba767)


You are always advised not to mess with the registry but to change which internal icon is associated with a file type you can use any included so for mobi I can change it to ePub like this by replace 1 with 2 and **save and refresh icons**
![image](https://github.com/user-attachments/assets/1a319503-ec2b-4e5a-ab2c-bf9b11656791)

![image](https://github.com/user-attachments/assets/269d5acc-513e-4e3d-a93f-d47da7513c62)

I can use an icon from any other source (such as an icon library or a different exe). So the icon is assignable as you wish inside the registry.
![image](https://github.com/user-attachments/assets/6008995d-5ed0-4ddd-af2f-fd18c289c180)
